src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,98 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useCallback, useContext, useEffect, useState } from 'react';
+import { Person } from '../types';
+import { DispatchContext, StateContext } from '../store/GlobalContextProvider';
+import { peopleService } from '../utils/peopleService';
 
 export const PeoplePage = () => {
+  const [loading, setLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const { people, searchResultCount } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const hasError = !loading && errorMessage;
+  const hasNoPeople = !loading && !errorMessage && !people.length;
+  const showPeople = !loading && !hasError && !hasNoPeople;
+  const noResultsFound =
+    !loading && !hasError && !hasNoPeople && searchResultCount;
```

Comment:
The `noResultsFound` condition seems incorrect. It should check if `searchResultCount` is zero to determine if no results are found. Consider changing the condition to `!loading && !hasError && searchResultCount === 0`.